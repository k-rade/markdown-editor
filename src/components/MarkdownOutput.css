.dark .markdown-output {
    color: var(--color-gray-300);
}

h1,
h2,
h3,
h4,
h5,
h6 {
    color: var(--color-gray-900);
}

.dark h1,
.dark h2,
.dark h3,
.dark h4,
.dark h5,
.dark h6 {
    color: var(--color-gray-100);
}

blockquote {
    border-left: 2px solid var(--color-sky-400);
    padding: 0.5rem;
    margin: 0.5rem 0;
    background-color: hsla(199, 89%, 48%, 0.125);
}

.dark blockquote {
    border-color: var(--color-sky-500);
}

a {
    color: var(--color-sky-600);
}

a:hover {
    text-decoration: underline;
}

.dark a {
    color: var(--color-sky-400);
}

.dark table,
.dark td {
    border-color: var(--color-gray-500);
}

table {
    margin: 1rem 0;
    width: 100%;
    border: 1px solid var(--color-gray-200);
}

th {
    padding: 1rem;
    background-color: var(--color-gray-50);
}

.dark th {
    background-color: var(--color-gray-600);
}

td {
    padding: 0.5rem 1rem;
    border-top: 1px solid var(--color-gray-200);
}

.dark hr {
    border-color: var(--color-gray-700);
}

ul {
    margin-left: 1rem;
    list-style: disc;
}