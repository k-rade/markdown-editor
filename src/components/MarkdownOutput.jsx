import Markdown from "./Markdown";

import "./MarkdownOutput.css";

const MarkdownOutput = ({ markdown }) => {
  return (
    <div className="markdown-output">
      <Markdown markdown={markdown} />
    </div>
  );
};

export default MarkdownOutput;
