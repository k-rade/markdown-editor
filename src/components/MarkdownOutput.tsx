import ReactMarkdown from "react-markdown";
import remarkGfm from "remark-gfm";
import { Prism as SyntaxHighlighter } from "react-syntax-highlighter";
import { atomDark } from "react-syntax-highlighter/dist/esm/styles/prism";
import "./MarkdownOutput.css";

interface Props {
  markdown: string;
}

const MarkdownOutput = ({ markdown }: Props): JSX.Element => {
  return (
    <div className="w-full max-w-5xl m-auto sm:mt-8 sm:px-8 h-[calc(100vh-8rem)] sm:h-[calc(100vh-6rem)]">
      <div className="markdown-output overflow-y-auto shadow-lg bg-white h-full p-4 sm:p-8 rounded-md dark:bg-zinc-700">
        <ReactMarkdown
          children={markdown}
          remarkPlugins={[remarkGfm]}
          components={{
            code({ node, inline, className, children, ...props }) {
              const match = /language-(\w+)/.exec(className || "");
              return !inline && match ? (
                <SyntaxHighlighter
                  children={String(children).replace(/\n$/, "")}
                  style={atomDark}
                  language={match[1]}
                  PreTag="div"
                  spread={props}
                />
              ) : (
                <code className={className} {...props}>
                  {children}
                </code>
              );
            },
          }}
        />
      </div>
    </div>
  );
};

export default MarkdownOutput;
