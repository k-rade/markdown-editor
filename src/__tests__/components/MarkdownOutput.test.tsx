import { render } from "@testing-library/react";
import MarkdownOutput from "../../components/MarkdownOutput";

jest.mock("react-markdown", () => (props: any): JSX.Element => {
  return <div>{props.children}</div>;
});
jest.mock("remark-gfm", () => () => {});
jest.mock("react-syntax-highlighter/dist/esm/styles/prism", () => () => {});
jest.mock("../../util/markdownEditorUtils", () => ({
  downloadFile: jest.fn(),
}));

describe("MarkdownOutput", (): void => {
  it("should render", (): void => {
    const { baseElement } = render(
      <MarkdownOutput markdown="markdown content" />
    );

    expect(baseElement).toMatchSnapshot();
  });
});
