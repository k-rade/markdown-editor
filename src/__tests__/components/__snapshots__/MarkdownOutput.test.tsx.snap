// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MarkdownOutput should render 1`] = `
<body>
  <div>
    <div
      class="w-full max-w-5xl m-auto sm:mt-8 sm:px-8 h-[calc(100vh-8rem)] sm:h-[calc(100vh-6rem)]"
    >
      <div
        class="markdown-output overflow-y-auto h-full dark:bg-neutral-900"
      >
        <div>
          markdown content
        </div>
      </div>
    </div>
  </div>
</body>
`;
